### ISSUE TITLE
Please consider fetching users via accountsservice

### ISSUE THREAD

**Issue by agx** [samcday/phrog#22]
phrog currently uses the uzers crate to get the users. An alternative would be the `org.freedesktop.accounts` DBus interface which allows for additional properties like an avatar (via the `icon` property) to show a nicer greeter.

**Comment by samcday** [comment]
Yep, great suggestion, I'll definitely do this. I'm using the "mock users" feature of the `u(?![zs])ers` crate to support the `fakegreet` flow (*very* useful during development iterations). But of course, I'm going to need to introduce D-Bus mocking fixtures and infra to the local+CI tests before too long anyway.

**Comment by agx** [comment]
python-dbusmock has out of the box mock classes for many services. Idk if it's worth pulling in pyo3 though.

**Comment by samcday** [comment]
> python-dbusmock has out of the box mock classes for many services. Idk if it's worth pulling in pyo3 though.

[bah](https://github.com/martinpitt/python-dbusmock/blob/812dc61498e021bec2ff3978aa812ac11a1db855/NEWS#L8) :(

**Comment by agx** [comment]
If there's a consumer they could be added back one can also easily ship custom templates and still use the python-dbusmock mockery)

**Comment by samcday** [comment]
Aye - I recall seeing you send a upower template or something recently upstream over there.

Anyway, it's not a big deal since I want to learn how to drive dbusmock properly.

**Comment by samcday** [comment]
I decided against dbusmock. AFAICT the code to spawn a separate system bus is only usable from Python itself via the test fixtures / methods (i.e not exposed via the CLI), so I have to do that myself anyway. Since I'm already pulling in zbus for the client side, and the service side seems pretty simple, I'm going that route instead.

**Comment by samcday** [comment]
The initial work is merged, the test recording from 92d46ea9188d7e88cc3f726a779cc90648838626 is looking very snazzy with Phoshi gracing it.

I think if I just casually drop this here, I can also (somewhat) stable link it in an embed from the README? I will try that next:

https://github.com/user-attachments/assets/8ce115c4-68c0-4e46-a58e-21d18ca73f8a

**Comment by agx** [comment]
> Video can't be played because it's corrupt

:cry:

**Comment by agx** [comment]
That plays fine, thanks! Looks nice!

**Comment by samcday** [comment]
Cool, and thanks :D

I guess I'll have to add a CI postprocessing step that flattens the mov down into a `.webp` or maybe even just a `.gif` so it can be more portably shared in Markdown. Something to consider if/when I finish #18 :innocent:

**Comment by agx** [comment]
I noticed (and watching the video reminded me) that that the UI lacks a bit of border radius for user selection / session selection (as in https://gitlab.com/mobian1/phog/-/raw/main/.assets/screenshots/phog-1.png) . I can do a separate issue if that helps. 

Likely known but just in case: It's possible to run gtk-inspector to test css changes on the fly in the running instance with the inspector window showing up outside the nested session: https://world.pages.gitlab.gnome.org/Phosh/phosh/gettingstarted.html#debugging

**Comment by samcday** [comment]
I grabbed some random CSS rules from phog and things are indeed looking a little more rounded in #38 :smile: I also figured out how to fire up Inspector which makes it much easier and quicker to iterate CSS changes, thanks :+1:

**Comment by samcday** [comment]
Before I consider this done, I still want to:

* Run a glib local async loop on the `AccountsProxy` and create/remove rows when the user signals fire (thus removing the current `AccountsProxyBlocking` usage)
* Move the user row setup from `UserSessionPage` to an individual widget
* Delegate the mapping of user properties to a glib local async loop that starts from user row widget constructor (thus removing `UserProxyBlocking` usage)

This ensures that all the DBus roundtrips and Pixbuf happens off the main constructor, ensuring that `PhoshShell` sends the ready signal as soon as possible.

**Comment by samcday** [comment]
With #50 merged I'm declaring this as "done". The DBus interactions all take place off the main thread as much as possible, and the relevant properties are watched for changes. As a bonus, the profile picture is watched for changes too, both from the dbus property changing, and a `Gio.FileMonitor` on the file itself (AccountsService just keeps changing the one file on my machine).


https://github.com/user-attachments/assets/ca0290e7-05f5-468d-bb03-7e07cb70318d

**Comment by samcday** [comment]
@agx I think I figured out why the recordings I was sharing with you earlier weren't working: wf-recorder is defaulting to VP9 codec when I do local recordings on my machine :sweat_smile: Hopefully you can easily view the demo above (there's no CI recordings or test cases for the bindings behaviour).

Thanks for suggesting this idea. Let me know if there's any further follow-ups on this that you'd like to see. Or anything further follow-ups in general, really :)

**Comment by agx** [comment]
Reads great, thanks!  Regarding the video:  I see "Video can't be played because the file is corrupt again". :cry:

**Comment by samcday** [comment]
I'm quite curious why your user agent isn't playing h264. I figured that should be the most widely compatible codec on the web. Maybe I should be opting for another? :thinking:

Anyway great that you got to see it :+1:

**Comment by agx** [comment]
For fedi posts I usually convert to webm (and UA also usually plays h264 fine)

**Comment by samcday** [comment]
> For fedi posts I usually convert to webm (and UA also usually plays h264 fine)

Ah-ha, so this works I guess?

[demo22.webm](https://github.com/user-attachments/assets/0423efb8-c02b-4a24-8ff1-322fa367342f)

**Comment by agx** [comment]
yup, that ones plays nicely
